### PR TITLE
build: Use ubi as default based image for che-server build

### PR DESCRIPTION
**Pull request by skabashnyuk**
### What does this PR do?
Use ubi as default based image for che-server build.

### Screenshot/screencast of this PR
n/a

### What issues does this PR fix or reference?
To ensure the latest java updates
To align with che-operator base image.

### How to test this PR?
n/a

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Review by sparkoo — APPROVED** [review]
please update `adopt` to `temurin`. It was renamed recently and I'm not sure if `adopt` will receive latest updates.

**Review by nickboldt — APPROVED** [review]
not sure about the GHA changes (seems OK) but the rename of rhel.Dockerfile to Dockerfile so we have the same content in Che and CRW? :+1: 

Can we merge this in time for 7.44, including a PR against https://github.com/redhat-developer/codeready-workspaces-images/blob/crw-2-rhel-8/codeready-workspaces/build/scripts/sync.sh#L88-L89 to adapt to the new upstream source dockerfile?

**Comment by nickboldt** [comment]
Downstream, you'll want to submit a PR against this folder so that we sync from Dockerfile instead of rhel.Dockerfile https://github.com/redhat-developer/codeready-workspaces-images/tree/crw-2-rhel-8/codeready-workspaces --> https://github.com/redhat-developer/codeready-workspaces-images/blob/crw-2-rhel-8/codeready-workspaces/build/scripts/sync.sh#L88-L89

Also... while you're refactoring stuff, could you move content into `build/dockerfiles/` instead of  `dockerfiles/che/` ? Other Che projects do that... so this one is a bit inconsistent. Just an idea. :D

**Comment by nickboldt** [comment]
If you want to pin a specific version of OpenJDK 11, you can do something like:

`yum -y install java-11-openjdk-11.0.14.0.9-2.el8_5`

File: dockerfiles/che/Dockerfile [hunk 1]
```diff
@@ -1,18 +1,37 @@
...
-FROM docker.io/adoptopenjdk/openjdk11:jre-11.0.11_9
-ENV LANG=C.UTF-8
-RUN echo "%root ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers && rm -rf /tmp/*
-EXPOSE 8000 8080
+# https://access.redhat.com/containers/?tab=tags#/registry.access.redhat.com/ubi8-minimal
+FROM registry.access.redhat.com/ubi8-minimal:8.5-230
+USER root
+ENV CHE_HOME=/home/user/eclipse-che
+ENV JAVA_HOME=/usr/lib/jvm/jre
+RUN microdnf install java-11-openjdk-headless tar gzip shadow-utils findutils && \
```
Comment (nickboldt) [inline dockerfiles/che/Dockerfile:17]: Could pin to a specific runtime rpm with `microdnf install java-11-openjdk-headless-11.0.14.0.9-2.el8_5`

File: dockerfiles/che/Dockerfile [hunk 2]
```diff
@@ -1,18 +1,37 @@
...
+    microdnf update -y gnutls && \
```
Comment (nickboldt) [inline dockerfiles/che/Dockerfile:18]: why only update gnutls? Why not pull all updates, THEN install `microdnf install -y java-11-openjdk-headless-11.0.14.0.9-2.el8_5` ?

Comment (skabashnyuk) [reply dockerfiles/che/Dockerfile:18]: can you make an example of a change you want to make?

**Pull request merged**